commit c987d602176cb06f7d295d57e7f19b8d3dffc6e1 (refs/changes/38/2438/5)
Author: Jan Wyszynski <jan.wyszynski@joyent.com>
Date:   2017-09-30T00:10:32+00:00 (2 years ago)
    
    MANTA-3363 Some MPU operations don't log shard information in muskie audit entry

diff --git a/lib/audit.js b/lib/audit.js
index f45fa40..2d547d5 100644
--- a/lib/audit.js
+++ b/lib/audit.js
@@ -271,6 +271,8 @@ function auditLogger(options) {
         obj.sharksContacted = req.sharksContacted;
         obj.entryShard = req.entryShard;
         obj.parentShard = req.parentShard;
+        obj.uploadRecordShard = req.uploadRecordShard;
+        obj.finalizingRecordShard = req.finalizingRecordShard;
 
         if (req.route) {
             obj.route = req.route.name;
diff --git a/lib/uploads/common.js b/lib/uploads/common.js
index f432788..6af7246 100644
--- a/lib/uploads/common.js
+++ b/lib/uploads/common.js
@@ -551,6 +551,10 @@ function loadUploadRecord(upload, cb) {
             assert.ok(wrap._etag);
             record.toSave._etag = wrap._etag;
 
+            if (wrap && wrap._node) {
+                upload.req.uploadRecordShard = wrap._node.pnode;
+            }
+
             cb(null, record.loaded.upload);
         }
     });
@@ -568,12 +572,17 @@ function loadFinalizingRecord(upload, cb) {
         requestId: upload.req.getId()
     };
 
-    upload.req.moray.getFinalizingMetadata(options, function (err, md) {
+    upload.req.moray.getFinalizingMetadata(options, function (err, md, wrap) {
         if (err) {
             cb(err);
         } else {
             assert.ok(md);
             record.loaded = md;
+
+            if (wrap && wrap._node) {
+                upload.req.finalizingRecordShard = wrap._node.pnode;
+            }
+
             cb(null, record.loaded);
         }
     });
